cassandra/cluster.py

```diff
@@ -2073,6 +2097,24 @@ def _handle_results(success, result):
             log.debug("[control connection] Fetched user type info for %s.%s, rebuilding metadata", keyspace, usertype)
             types_result = dict_factory(*types_result.results) if types_result.results else {}
             self._cluster.metadata.usertype_changed(keyspace, usertype, types_result)
+        elif function:
+            # user defined function within this keyspace changed
+            where_clause = " WHERE keyspace_name = '%s' AND function_name = '%s' AND signature = [%s]" \
+                           % (keyspace, function.name, ','.join("'%s'" % t for t in function.type_signature))
```

Comment:
I was mostly just using the information that comes in the schema_change event. I think it's a fine idea to make it optional for the ``Cluster.refresh_schema`` API. I've added a note to the ticket I have to split this up: https://datastax-oss.atlassian.net/browse/PYTHON-291